plugins/woocommerce/includes/emails/class-wc-email-cancelled-order.php

```diff
@@ -97,6 +97,10 @@ public function trigger( $order_id, $order = false ) {
 				$this->placeholders['{order_billing_full_name}'] = $this->object->get_formatted_billing_full_name();
 			}
 
+			if ( $this->block_email_editor_enabled ) {
+				$this->personalizer->set_context( $this->prepare_context_data( $this->personalizer->get_context() ) );
+			}
+
```

Comment:
I think we try to avoid modifying individual emails as much as possible. In this case we repeat the same two lines in every email type class. This will add a requirement that every plugin that adds email type will need to add these two lines.

Could we move it to `WC_Email::send` ? It receives the subject, so we could personalize it directly in the method or add a new filter there. I think it would be better to avoid changing every sinle WC_Email. What do you think?

Reply:
> Could we move it to WC_Email::send ? It receives the subject, so we could personalize it directly in the method or add a new filter there. I think it would be better to avoid changing every sinle WC_Email. What do you think?

I am not sure about this approach. Because the replacement would be used only in the case we send the email through this method. But what about a preview? For example, sending a preview is managed in the PHP email editor package through `wp_mail`. Another use case can be HTML preview.  I prefer keeping the setting context for personalization tags in the same place as it was for the old placeholders. Maybe it would be better to chat about it if you want.